public/editor/scripts/project-list.js

```diff
@@ -22,6 +22,18 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
   var username = encodeURIComponent($("#project-list").attr("data-username"));
   var locale = $("html")[0].lang;
   var queryString = window.location.search;
+  var isLocalStorageAvailable = !!(window.localStorage);
+  var favorites;
+  if(isLocalStorageAvailable){
+    try {
+      favorites = JSON.parse(localStorage.getItem("project-favorites"));
+      if(favorites === null) {
```

Comment:
you don't need this check, instead,
set `var favorites` to `var favorites = []`